tests/tests/test_monitor_client.py

```diff
@@ -1121,8 +1121,10 @@ def test_monitorclient_logs_and_services(self, monitor_commercial_setup_no_clien
         _, messages = get_and_parse_email_n(
             monitor_commercial_setup_no_client, user_name, 4
         )
-        assert messages_count >= 4
-        for m in [messages[-1], messages[-2]]:
+        messages_count = len(messages)
+        assert messages_count == 5
```

Comment:
I think we are abusing this `message_count` variable. It is being re-defined in many places and prone to errors. I think we should remove it completely and do only:
```suggestion
        assert len(messages) == 5
```

We can also clean-up this messy block above where `message_count` is being redefined without even checking, and we have two asserts for the same thing:
![image](https://user-images.githubusercontent.com/3168644/159648814-30d2f5c3-536c-4061-9010-061017cbff43.png)

Reply:
cleaned-fixed-squashed. thanks.